src/main/java/org/folio/ebsconet/service/NotesService.java

```diff
@@ -28,33 +31,37 @@ public Note createNote(Note note) {
     log.info("Creating customer note: {}", note);
     return notesClient.postNote(note);
   }
+
   public void deleteNote(String id) {
     log.info("Deleting customer note by id {}", id);
-     notesClient.deleteNote(id);
+    notesClient.deleteNote(id);
   }
 
   public void linkCustomerNote(MappingDataHolder mappingDataHolder) {
     log.debug("Link customer note to poLine");
-    String generalNoteTypeId = noteTypeClient.getNoteTypesByQuery("name==" + GENERAL_NOTE_TYPE)
-        .get("noteTypes")
-        .get(0)
-        .get("id")
-        .asText();
+    JsonNode generalNote = noteTypeClient.getNoteTypesByQuery("name==" + GENERAL_NOTE_TYPE)
+      .get("noteTypes")
+      .get(0);
+    if (generalNote == null) {
+      throw new ResourceNotFoundException("Note not found with name==" + GENERAL_NOTE_TYPE);
+    }
+    String generalNoteTypeId = generalNote
+      .get("id")
+      .asText();
     var note = this.getNoteByPoLineId(generalNoteTypeId, mappingDataHolder.getCompositePoLine().getId());
     log.info("Note {}, generalNotTypeId: {}", note, generalNoteTypeId);
-    if (mappingDataHolder.getEbsconetOrderLine().getCustomerNote() != null
-      && !mappingDataHolder.getEbsconetOrderLine().getCustomerNote().isEmpty()) {
+    String customerNote = mappingDataHolder.getEbsconetOrderLine().getCustomerNote();
+    if (customerNote != null && !customerNote.isEmpty()) {
```

Comment:
better to use StringUtils.isNotBlank here

Reply:
fixed